packages/design-system/src/components/N8nMenu/Menu.vue

```diff
@@ -1,22 +1,83 @@
 <template>
-	<el-menu
-		:defaultActive="defaultActive"
-		:collapse="collapse"
-		:router="router"
-		:class="['n8n-menu', $style[type + (light ? '-light' : '')]]"
-		v-on="$listeners"
-	>
-		<slot></slot>
-	</el-menu>
+	<div :class="{
+		['menu-container']: true,
+		[$style.container]: true,
+		[$style.menuCollapsed]: collapsed
+	}">
+		<div v-if="$slots.header" :class="$style.menuHeader">
+			<slot name="header"></slot>
+		</div>
+		<div :class="$style.menuContent">
+			<div :class="{[$style.upperContent]: true, ['pt-xs']: $slots.menuPrefix === undefined  }">
+				<div v-if="$slots.menuPrefix" :class="$style.menuPrefix">
+					<slot name="menuPrefix"></slot>
+				</div>
+				<el-menu
+					:defaultActive="defaultActive"
+					:collapse="collapsed"
+					v-on="$listeners"
+				>
+					<n8n-menu-item
+						v-for="item in upperMenuItems"
+						:key="item.id"
+						:item="item"
+						:compact="collapsed"
+						:popperClass="$style.submenuPopper"
+						:tooltipDelay="tooltipDelay"
+						:mode="mode"
+						:activeTab="activeTab"
+						@click="onSelect"
+					/>
+				</el-menu>
+			</div>
+			<div :class="{[$style.lowerContent]: true, ['pb-xs']: $slots.menuSuffix === undefined}">
```

Comment:
just a nitpick (on code style)
```suggestion
			<div :class="{[$style.lowerContent]: true, 'pb-xs': !$slots.menuSuffix}">
```